On Cocoa, the SCI_POSITIONFROMPOINTCLOSE returns -1 when the editor has been
scrolled - bug 100487.
diff -r baaa425295db src/Editor.cxx
--- src/Editor.cxx	Sun Sep 22 12:59:01 2013 +1030
+++ src/Editor.cxx	Tue Sep 24 17:42:24 2013 -0700
@@ -542,6 +543,15 @@
 	RefreshStyleData();
 	if (canReturnInvalid) {
 		PRectangle rcClient = GetTextRectangle();
+
+#if MACOSX
+		// KOMODO: Fix rcClient co-ords to be relative (not absolute).
+		rcClient.bottom -= rcClient.top;
+		rcClient.top = 0;
+		rcClient.right -= rcClient.left;
+		rcClient.left = 0;
+#endif
+
 		if (!rcClient.Contains(pt))
 			return SelectionPosition(INVALID_POSITION);
 		if (pt.x < vs.textStart)
